accessible/toolboxview.component.js

```diff
@@ -0,0 +1,141 @@
+/**
+ * Blockly Demos: AccessibleBlockly
+ *
+ * Copyright 2016 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Component that details how a toolbox is rendered
+ * in AccessibleBlockly. Also handles any interactions with the toolbox.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+var blocklyApp = blocklyApp || {};
+
+blocklyApp.ToolboxView = ng.core
+.Component({
+  selector: 'toolbox-view',
+  template: `
+<h3 #toolboxTitle id='blockly-toolbox-title'>Toolbox</h3>
+<ol #tree id='blockly-toolbox-tree' *ngIf='makeArray(sightedToolbox) && makeArray(sightedToolbox).length > 0' tabIndex='0' role='group' class='blocklyTree' [attr.aria-labelledby]='toolboxTitle.id' (keydown)='treeService.keyHandler($event, tree)'>
+{{treeService.setActiveAttribute(tree)}}
+<li #parent *ngIf='toolboxHasCategories' role='treeitem' aria-level='1' aria-selected=false *ngFor='#category of makeArray(sightedToolbox); #i=index'>
+  <label #name>{{category.attributes.name.value}}</label>
+  {{labelCategory(name, i, tree)}}
+  <ol  role='group' *ngIf='getToolboxWorkspace(category).topBlocks_.length > 0'>
+    <toolbox-tree-view *ngFor='#block of getToolboxWorkspace(category).topBlocks_' [level]=2 [block]='block' [displayBlockMenu]='true' [clipboardService]='sharedClipboardService'></toolbox-tree-view>
+    {{addClass(parent, 'blocklyHasChildren')}}
+  </ol>
+</li>
+<div *ngIf='!toolboxHasCategories'>
+  <toolbox-tree-view *ngFor='#block of getToolboxWorkspace(toolboxCategories[0]).topBlocks_; #i=index' [level]=1 [block]='block' [displayBlockMenu]='true' [clipboardService]='sharedClipboardService' [index]='i' [tree]='tree'></toolbox-tree-view>
+</div>
+</ol>
+  `,
+  directives: [blocklyApp.ToolboxTreeView],
+})
+.Class({
+  constructor: [blocklyApp.TreeService, function(_service) {
+    console.log('toolbox constructor');
+    var _this = this;
+    var xhttp = new XMLHttpRequest();
+    xhttp.onreadystatechange = function() {
+      if (xhttp.readyState == 4 && xhttp.status == 200) {
```

Comment:
Good to know. This probably will all go away when I move the toolbox into the html instead of having it in a separate XML file.